superset/views/core.py

```diff
@@ -1260,9 +1255,11 @@ def explore(self, datasource_type=None, datasource_id=None):
             flash(DATASOURCE_MISSING_ERR, 'danger')
             return redirect(error_redirect)
 
-        if not security_manager.datasource_access(datasource):
+        if config.get('ENABLE_ACCESS_REQUEST') and (
```

Comment:
Why was this check added?

Reply:
A check is done at a lower point (in explore_json) This check here is strictly for redirecting users who don't have access to the access request page.